src/main/java/io/passport/server/service/ModelDeploymentService.java

```diff
@@ -0,0 +1,126 @@
+import org.springframework.stereotype.Service;
+
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Service class for model deployment management.
+ */
+@Service
+public class ModelDeploymentService {
+
+    /**
+     * ModelDeployment repo access for database management.
+     */
+    private final ModelDeploymentRepository modelDeploymentRepository;
+
+    /**
+     * Deployment Environment service access.
+     */
+    private final DeploymentEnvironmentService deploymentEnvironmentService;
+
+
+    @Autowired
+    public ModelDeploymentService(ModelDeploymentRepository modelDeploymentRepository, DeploymentEnvironmentService deploymentEnvironmentService) {
+        this.modelDeploymentRepository = modelDeploymentRepository;
+        this.deploymentEnvironmentService = deploymentEnvironmentService;
+    }
+
+    /**
+     * Return all model deployments
+     * @return
+     */
+    public List<ModelDeployment> getAllModelDeployments() {
+        return modelDeploymentRepository.findAll();
+    }
+
+
+
+    /**
+     * Find a model deployment by deploymentId
+     * @param deploymentId ID of the model deployment
+     * @return
+     */
+    public Optional<ModelDeployment> findModelDeploymentByDeploymentId(Long deploymentId) {
+        return modelDeploymentRepository.findById(deploymentId);
+    }
+
+
+    /**
+     * Save a model deployment
+     * @param modelDeployment model deployment to be saved
+     * @return
+     */
+    public ModelDeployment saveModelDeployment(ModelDeployment modelDeployment) {
+        // Set the creation and last update timestamps
+        Instant now = Instant.now();
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss").withZone(ZoneId.systemDefault());
+        String formattedTime = formatter.format(now);
+        Timestamp timestamp = Timestamp.valueOf(formattedTime);
+        modelDeployment.setCreatedAt(timestamp);
+        modelDeployment.setLastUpdatedAt(timestamp);
+
+        return modelDeploymentRepository.save(modelDeployment);
+    }
+
+    /**
+     * Update a model deployment
+     * @param deploymentId ID of the model deployment
+     * @param updatedModelDeployment model deployment to be updated
+     * @return
+     */
+    public Optional<ModelDeployment> updateModelDeployment(Long deploymentId, ModelDeployment updatedModelDeployment) {
+        Optional<ModelDeployment> oldModelDeployment = modelDeploymentRepository.findById(deploymentId);
+        if (oldModelDeployment.isPresent()) {
+            ModelDeployment modelDeployment = oldModelDeployment.get();
+
+            // set last updated timestamp
+            Instant now = Instant.now();
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss").withZone(ZoneId.systemDefault());
```

Comment:
You do not need to format time. You can pass Instant object directly to database.